# COGS 108 - Final Project: Vehicle Stops in San Diego

### Group Members: 
- Louise Xu (A12650425)
- David Tay
- Jeffery Lu

### Goal of Project:
- To determine whether there is a correlation between the frequency of vehicle stops in San Diego and the time and locations at which they take place
- To examine possible reasons for the correlations, if any exist 

### Research Question
- Are there more vehicle stops at certain times and places than others? Does the frequency of vehicle stops vary significantly between different seasons?

### Hypothesis
- There will most likely be more traffic violations during periods of time with high traffic density (e.g. rush hour), during weekdays (when there are more people on the road who are driving to work), and during holiday seasons (since there would be more tourists and people visting family). On a broader scale, given that the population of San Diego is increasing, it reasonable to say that the number of traffic violations is also increasing at a similar rate. 
- We can also guess that the number of vehicle stops in more populated areas is greater than those in less populated areas.


In [1]:
# Import packages
%matplotlib inline

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import patsy
import statsmodels.api as sm
from scipy.stats import ttest_ind

In [6]:
# Import each vehicle stops data file into its own DataFrame
vs_14 = pd.read_csv('vehicle_stops_2014_datasd.csv')
vs_15 = pd.read_csv('vehicle_stops_2015_datasd.csv')
vs_16 = pd.read_csv('vehicle_stops_2016_datasd.csv')
vs_17 = pd.read_csv('vehicle_stops_2017_datasd.csv')

# Join the data for each year together into one DataFrame
# Note: The file for vs_17 contained two more columns (both of them empty and unnamed) than the datasets for the other years 
# These two columns were deleted on Microsoft Excel in order to successfully merge the four datasets together
vs_all = pd.concat([vs_14, vs_15, vs_16, vs_17], ignore_index=True)

#Import police regions into a DataFrame
sd_neigh = pd.read_csv('pd_beat_neighborhoods_datasd.csv')

In [7]:
# Let's check to see how many observations are in each dataset
# Note: vs_17 is year to date data from 2017, so there is data up to May of 2017 (the period in which this analysis is conducted)
print(vs_14.shape)
print(vs_15.shape)
print(vs_16.shape)
print(vs_17.shape)

(144164, 15)
(115422, 15)
(103051, 15)
(28362, 15)


We can see here that the number of vehicle stops is actually *decreasing* each year between 2014 and 2016. Also, while we can not take the 2017 data into direct consideration because we only have data up to May, we can predict the total number of vehicle stops at the end of this year based on our current data (28362 stops) divided by the five months that have already passed and then multiplied by the twelve months of the year, which results in (28362)(12)/5 = 68068.8 stops. We can round up to the nearest ten thousandsth and estimate that approximately 70000 vehicle stops will happen in the year of 2017. This still follows the trend of decreasing vehicle stops in San Diego.